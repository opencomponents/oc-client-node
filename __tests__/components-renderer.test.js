'use strict';

const ComponentsRenderer = require('../src/components-renderer');
jest.mock('../src/get-components-data');

test('ComponentsRenderer (factory) is a function', () => {
  expect(typeof ComponentsRenderer).toEqual('function');
});

test('componentsRenderer (instance) is a function api', () => {
  const config = {};
  const componentsRenderer = new ComponentsRenderer(config);
  expect(typeof componentsRenderer).toEqual('function');
});

test('componentsRenderer happy path', () => {
  const config = {};
  const componentsRenderer = new ComponentsRenderer(config);
  const components = [
    {
      name: 'good',
      version: '1.0.0'
    }
  ];
  const callback = jest.fn();
  componentsRenderer(components, null, callback);
  expect(callback).toHaveBeenCalled();
  expect(callback.mock.calls).toMatchSnapshot();
});

test('componentsRenderer has errors', () => {
  const config = {
    components: {
      'bad': '1.0.0'
    }
  };
  const componentsRenderer = new ComponentsRenderer(config);
  const components = [
    {
      name: 'bad'
    }
  ];
  const callback = jest.fn();
  componentsRenderer(components, null, callback);
  expect(callback).toHaveBeenCalled();
  expect(callback.mock.calls).toMatchSnapshot();
});
