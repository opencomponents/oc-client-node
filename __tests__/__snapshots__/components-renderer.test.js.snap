// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`componentsRenderer happy path 1`] = `
Array [
  Array [
    null,
    Array [
      undefined,
    ],
  ],
]
`;

exports[`componentsRenderer has errors 1`] = `
Array [
  Array [
    Array [
      "something bad happened",
    ],
    Array [
      undefined,
    ],
  ],
]
`;
